mne/io/cnt/cnt.py

```diff
@@ -0,0 +1,229 @@
+        input_fname = path.abspath(input_fname)
+        info, cnt_info = _get_cnt_info(input_fname, read_blocks)
+        last_samps = [cnt_info['n_samples'] - 1]
+        super(RawCNT, self).__init__(
+            info, preload, filenames=[input_fname], raw_extras=[cnt_info],
+            last_samps=last_samps, orig_format='int',
+            verbose=verbose)
+
+    @verbose
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Take a chunk of raw data, multiply by mult or cals, and store"""
+        n_channels = self.info['nchan'] - 1
+        n_samples = self._raw_extras[0]['n_samples']
+        channel_offset = self._raw_extras[0]['channel_offset']
+        baselines = self._raw_extras[0]['baselines']
+        event_offset = self._raw_extras[0]['event_offset']
+        n_bytes = 2
+        # The data is divided into blocks of samples / channel.
+        # channel_offset determines the amount of successive samples.
+        # Here we use sample offset to align the data because start can be in
+        # the middle of these blocks.
+        s_offset = start % channel_offset
+
+        with open(self._filenames[fi], 'rb', buffering=0) as fid:
+            fid.seek(event_offset)
+            event_type = np.fromfile(fid, dtype='<i1', count=1)[0]
+            event_size = np.fromfile(fid, dtype='<i4', count=1)
+            if event_type == 1:
+                bytes = 8
+            elif event_type == 2:
+                bytes = 19
+            else:
+                raise IOError('Unexpected event size.')
+            n_events = event_size / bytes
+            events = list()
+            for i in range(n_events):
+                fid.seek(event_offset + 9 + i * bytes + 4)
+                offset = np.fromfile(fid, dtype='<i4', count=1)
+                events.append((offset - 900 - 75 * n_channels) /
+                              (n_channels * 2))
+
+            event_ch = np.zeros(n_samples)
+
+            for event in events:
+                event_ch[event - 1] = 1
+
+            fid.seek(900 + n_channels * (75 + (start - s_offset) * n_bytes))
+            data_ = np.empty([n_channels, n_samples])
+            # One extra sample set is read here to make sure the desired time
+            # window is covered by the blocks.
+            for sampleset in range(n_samples / channel_offset + 1):
+                if sampleset * channel_offset >= stop - start + s_offset:
+                    data_ = data_[:, s_offset:stop - start + s_offset]
+                    break
+                block = np.fromfile(fid, dtype='<i2',
+                                    count=n_channels * channel_offset)
+                block = block.reshape(n_channels, channel_offset, order='C')
+                block_start = sampleset * channel_offset
+                data_[:, block_start:block_start + channel_offset] = block
+            event_block = event_ch[start:stop]
+            data_ = np.vstack((data_, event_block))
```

Comment:
This will also cause a memcopy. Better to allocate the full matrix at the beginning like `data_full = np.empty(...)` and use views like `data_ = data_full[:n_ch_data]`